frame/composable-traits/src/dex.rs

```diff
@@ -37,12 +37,13 @@ pub trait Amm {
 		pool_id: Self::PoolId,
 		lp_amount: Self::Balance,
 		min_expected_amounts: BTreeMap<Self::AssetId, Self::Balance>,
+		is_single_asset: bool,
```

Comment:
Consider refactoring this into a more semantic enum? It's difficult to figure out what `true`/`false` mean at the call site

Reply:
I completely agree with you. I will refactor this when I add the asset selection functionality for depositing/withdrawing liquidity (now base asset is used by default). Ok?

Reply:
awesome! 🚀

Reply:
Fixed by 9a2720c6005c160cb68ae969e0dd404f6a29479a.

I will update the weights later and recheck the code again, because there are many places where changes have occurred.